app/uk/gov/hmrc/cdsreimbursementclaimfrontend/views/utils/BasisOfClaims.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cdsreimbursementclaimfrontend.views.utils
+
+import cats.implicits.catsSyntaxEq
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.claims.JourneyBindable
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.BasisOfClaim.{CorrectionToRiskClassification, DuplicateEntry, EvidenceThatGoodsHaveNotEnteredTheEU, IncorrectExciseValue, allClaimsTypes}
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.{BasisOfClaim, ClaimNorthernIrelandAnswer, DraftClaim, TaxCode}
+
+sealed trait BasisOfClaims {
```

Comment:
Looks like the perfect place for a typeclass.
What is wrong with you :-) ?

Reply:
Why would you use a typeclass approach for something trivial like this? 
Why just don't have a small overridable function inside of case class to form key for rendering? 
In the end this is the only difference between those types. The typeclass approach will overcomplicate things.